### PR TITLE
Add swoole_http_server request->getMethod()

### PR DESCRIPTION
**Pull request by luolaifa000**
add request->getMethod method to get header method and optimize the parsing request header method.

### CODE REVIEW

File: ext-src/swoole_http_request.cc [hunk 1]
```diff
@@ -102,68 +103,7 @@ static inline char *http_trim_double_quote(char *ptr, int *len) {
...
 }
 
-static sw_inline const char *http_get_method_name(int method) {
-    switch (method) {
-    case PHP_HTTP_GET:
-        return "GET";
-    case PHP_HTTP_POST:
-        return "POST";
-    case PHP_HTTP_HEAD:
-        return "HEAD";
```
Comment (huanghantao) [inline ext-src/swoole_http_request.cc:112]: Perhaps we could simply change the contents of this function to

```cpp
 return swoole_http_method_str(method)
```

File: ext-src/swoole_http_request.cc [hunk 2]
```diff
@@ -40,6 +40,7 @@ SW_EXTERN_C_END
 #include <brotli/encode.h>
 #endif
 
+
```
Comment (huanghantao) [inline ext-src/swoole_http_request.cc:43]: Unnecessary blank lines

File: tests/swoole_http_server/create_request.phpt [hunk 3]
```diff
@@ -47,5 +49,24 @@ $req2 = Request::create(['parse_cookie' => false]);
 Assert::eq($req2->parse($data . "\r\n"), strlen($data) + 2);
 Assert::null($req2->cookie);
 
+
+
+
```
Comment (huanghantao) [inline tests/swoole_http_server/create_request.phpt:54]: Unnecessary blank lines

File: thirdparty/swoole_http_parser.h [hunk 4]
```diff
@@ -102,6 +102,9 @@ enum swoole_http_method
   };
 
 
+
+
+
```
Comment (huanghantao) [inline thirdparty/swoole_http_parser.h:107]: Unnecessary blank lines

File: ext-src/swoole_http_request.cc [hunk 5]
```diff
@@ -1040,6 +982,15 @@ static PHP_METHOD(swoole_http_request, parse) {
     RETURN_LONG(swoole_http_requset_parse(ctx, str, l_str));
 }
 
+static PHP_METHOD(swoole_http_request, getMethod) {
+    http_context *ctx = php_swoole_http_request_get_and_check_context(ZEND_THIS);
+    if (UNEXPECTED(!ctx)) {
+        RETURN_FALSE;
+    }
+    char *method = (char *) swoole_http_method_str((ctx->parser).method);
```
Comment (huanghantao) [inline ext-src/swoole_http_request.cc:990]: May be we can use `const char *method`

File: ext-src/swoole_http_request.cc (hunk 1 above)
Comment (luolaifa000) [reply ext-src/swoole_http_request.cc:112]: done

File: ext-src/swoole_http_request.cc [hunk 6]
```diff
@@ -1040,6 +983,15 @@ static PHP_METHOD(swoole_http_request, parse) {
     RETURN_LONG(swoole_http_requset_parse(ctx, str, l_str));
 }
 
+static PHP_METHOD(swoole_http_request, getMethod) {
+    http_context *ctx = php_swoole_http_request_get_and_check_context(ZEND_THIS);
+    if (UNEXPECTED(!ctx)) {
+        RETURN_FALSE;
+    }
+    const char *method = (char *) http_get_method_name((ctx->parser).method);
```
Comment (huanghantao) [inline ext-src/swoole_http_request.cc:991]: `(char *)`这个转换应该可以删了，没必要转？

Comment (luolaifa000) [reply ext-src/swoole_http_request.cc]: thanks, done.

**Review by huanghantao — APPROVED** [review]

**Pull request merged**